src/types/Slot0.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+/** @dev Slot0 is a packed version of solidity structure. 
+ * Using the packaged version saves gas by not storing the structure fields in memory slots.
+ * 
+ * Fields:
+ * 
+ * The current price
+ * uint160 sqrtPriceX96;
+ * 
+ * The current tick
+ * int24 tick;
+ * 
+ * Protocol fee, expressed in hundredths of a bip, upper 12 bits are for 1->0, and the lower 12 are for 0->1
+ * the maximum is 1000 - meaning the maximum protocol fee is 0.1%
+ * the protocolFee is taken from the input first, then the lpFee is taken from the remaining input
+ * uint24 protocolFee;
+ * 
+ * Used for the lp fee, either static at initialize or dynamic via hook
+ * uint24 lpFee;
+ */
```

Comment:
imo worth writing in this comment about the unused bits. How many, where they are etc.

Reply:
Even a single line like

`24 bits empty | 24 bits lpFee | 24 bits protocolFee | 24 bits tick | 160 bits sqrtPriceX96`

i think would make this super clear what the layout is and the direction of the packing

Reply:
Added